spark-kernel-dsv2/src/main/java/io/delta/spark/dsv2/scan/batch/KernelSparkBatchScan.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (2025) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.spark.dsv2.scan.batch;
+
+import static java.util.Objects.requireNonNull;
+
+import io.delta.spark.dsv2.scan.KernelSparkScanContext;
+import java.io.Serializable;
+import org.apache.spark.sql.connector.read.Batch;
+import org.apache.spark.sql.connector.read.InputPartition;
+import org.apache.spark.sql.connector.read.PartitionReaderFactory;
+
+/** Spark Batch implementation backed by Delta Kernel Scan. */
+public class KernelSparkBatchScan implements Batch, Serializable {
```

Comment:
Sorry - why this class and also `KernelSparkScanContext`?

Reply:
(Serializable is added accidentally -- we don't need it)

Mainly for future use as KernelSparkScanContext need to be mutable (like pushing runtime filter)

https://github.com/apache/spark/blob/master/sql/catalyst/src/main/java/org/apache/spark/sql/connector/read/SupportsRuntimeV2Filtering.java


KernelSparkBatchScan  scan1 = Scan.toBatch // file after data skipping using static predicate set by user
Scan.filter() // mutate KernelSparkScanContext by adding new filter
KernelSparkBatchScan  scan2 = Scan.toBatch // file after data skipping using dynamic predicate produce by planner